encryptionkeyrotation/README.md

```diff
@@ -0,0 +1,95 @@
+# CSI Addons Operation: EncryptionKeyRotation
+
+## Terminology
+
+| Term     | Definition                                                                            |
+| -------- | ------------------------------------------------------------------------------------- |
+| VolumeID | The identifier of the volume generated by the plugin.                                 |
+| CO       | Container Orchestration system that communicates with plugins using CSI service RPCs. |
+| SP       | Storage Provider, the vendor of a CSI plugin implementation.                          |
+| RPC      | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).         |
+
+## Objective
+
+Define an extension to the CSI specification that will enable storage vendors
+(SP) to develop controllers/plugins that can rotate the Encryption Keys (KEK)
+of existing volumes that are encrypted.
+
+### Goals in MVP
+
+The new extension will define a procedure that
+
+- can be called for existing encrypted volumes
+- functions for Block mode volumes
```

Comment:
Filesystem encryption is a thing too. ext4 with fscrypt can be used. Or, in case of CephFS it supports encryption as well.

Reply:
> in case of CephFS it supports encryption as well

We are only supporting RBD volumes right now. CephFS support shall be added later.

Reply:
> We can rotate keys for rbd filesystem access mode too.

Ack.

Reply:
> > in case of CephFS it supports encryption as well
> 
> We are only supporting RBD volumes right now. CephFS support shall be added later.

This is a specification of the protocols and procedures that need to be implemented by CSI drivers. Ceph-CSI is one of the drivers which can choose how to implement appropriately it for RBD, CephFS and NFS. The specification needs to be complete for all types of CSI volumes.

Reply:
Ack.